### ISSUE TITLE
Duplicated StackExchange notifications

### ISSUE THREAD

**Issue by xurizaemon** [xurizaemon/bolts#26]
Follows from #14 

Hubot double-posts some StackExchange questions, which stands out especially badly during quiet periods in Town Square.

![screenshot_20180510-044313](https://user-images.githubusercontent.com/105608/39828040-c9a5c426-540d-11e8-806d-9b77736ad37c.jpg)

We need to:

- [ ] work out why the poller detects the SE feed as having changed when the latest post hasn't - suspect this relates to question edits
- [ ] account for this and/or detect that the previous post on this feed was identical
- [ ] not do that!

xurizaemon/hubot-rss-poller2 is a fork of a module [published on NPM](https://www.npmjs.com/package/hubot-rss-poller2) - I'm not sure if there's a repo available for contributing back.

(@gobias-nz if you comment with a link to repo I'd love to contribute back fixes like this - looks like the published version on NPM is babel-ified and it would be more useful to contribute to the original code? Would love you to publish that as well!)

**Comment by joemcl** [comment]
Hey @xurizaemon see this - http://npminstall.org/Install-hubot-rss-poller2-using-npm/ - and this - https://github.com/KanoYugoro/hubot-rss-poller - @KanoYugoro any help with tracking down a repo for hubot-rss-poller2 ?

**Comment by xurizaemon** [comment]
Actually Joe, Kano's looks like it's what Gobias based their work off, so I will see if we can switch to using that - seems like the original repo is active while hubot-rss-poller2 is not updated since 0.0.5 on npm. Good find :)

**Comment by joemcl** [comment]
Yeah thanks though was wondering why Kano posted the download and install for poller2 rathern than poller...

**Comment by joemcl** [comment]
Anyway, cool

**Comment by joemcl** [comment]
Maybe poller2 is just v 3 of poller in Kanos repo....

**Comment by xurizaemon** [comment]
Current status: I switched to [xurizaemon/hubot-rss-poller@build](https://github.com/xurizaemon/hubot-rss-poller/tree/build) which is my fork of KanuYugoro/hubot-rss-poller. It's not working yet, so the dupe posts are fixed. Will revisit "soon" :)

**Comment by xurizaemon** [comment]
> Yeah thanks though was wondering why Kano posted the download and install for poller2 rather than poller...

I don't understand this comment @joemcl ?

**Comment by joemcl** [comment]
Kano/Anson Wayman is the author of poller, from which poller2 is forked, apparently. The link [here](http://npminstall.org/Install-hubot-rss-poller2-using-npm/) from him is to install poller2, the fork that someone else modified, via npm, rather than his original poller. However, I just discovered that in fact he posted links to install both versions, on the same day, see [here](http://npminstall.org/?s=rss-poller), via npm. Doesn't matter, just thought it was curious.

**Comment by xurizaemon** [comment]
Aha, yeah I believe it goes like this -

* npminstall.org is largely autogenerated and shows a templated set of instructions for any module on https://www.npmjs.com/ - those posts aren't by Anson, as you suspected
* Anson wrote hubot-rss-poller and published the source as that module to npmjs.com
* There's a build step required to *use* that module in a hubot, because the entry point in package.json is not included in the source (it's built with babel as part of publishing)
* Gobias-nz ran into this and worked around it by uploading a copy of Anson's work to npmjs.com, retaining the author in package.json
* I've now hit the same issue with my copy of Anson's work, but am dealing with it by [installing from a "build" branch](https://github.com/xurizaemon/bolts/blob/27501a6c4f006b1ed888a82406b0de3a17026143/package.json#L29) rather than shipping another copy of the module to npmjs.com

**Comment by joemcl** [comment]
OK, now I get it. Not sure what the purpose of that build step using babel is - I see [here](https://babeljs.io/) that babel is a Javascript compiler, I kinda know what a compiler does. Glad you know the workaround :).
